src/globus_cli/commands/gcs/endpoint/update.py

```diff
@@ -0,0 +1,244 @@
+)
+@click.option(
+    "--gridftp-control-channel-port",
+    type=IntOrNull(),
+    help="The TCP port which the Globus control channel should listen on.",
+)
+@click.option(
+    "--info-link",
+    type=str,
+    help=(
+        "An end-user-facing url for a webpage with more information about the endpoint."
+    ),
+)
+@click.option(
+    "--keywords",
+    type=CommaDelimitedList(),
+    help="[Searchable] A comma-separated list of search keywords.",
+)
+@click.option(
+    "--max-concurrency",
+    type=int,
+    help=(
+        "The endpoint network's custom max concurrency. Requires `network-use` be "
+        "set to `custom`."
+    ),
+)
+@click.option(
+    "--max-parallelism",
+    type=int,
+    help=(
+        "The endpoint network's custom max parallelism. Requires `network-use` be "
+        "set to `custom`."
+    ),
+)
+@click.option(
+    "--network-use",
+    type=click.Choice(["normal", "minimal", "aggressive", "custom"]),
+    help=(
+        "A control valve for how Globus will interact with this endpoint over the "
+        "network. If custom, you must also provide max and preferred concurrency "
+        "as well as max and preferred parallelism."
+    ),
+)
+@click.option(
+    "--organization",
+    type=str,
+    help="The organization which runs the server(s) represented by the endpoint.",
+)
+@click.option(
+    "--preferred-concurrency",
+    type=int,
+    help=(
+        "The endpoint network's custom preferred concurrency. Requires `network-use` "
+        "be set to `custom`."
+    ),
+)
+@click.option(
+    "--preferred-parallelism",
+    type=int,
+    help=(
+        "The endpoint network's custom preferred parallelism. Requires `network-use` "
+        "be set to `custom`."
+    ),
+)
+@click.option(
+    "--public/--private",
+    is_flag=True,
+    default=None,
+    help=(
+        "A flag indicating whether this endpoint is visible to all other Globus users. "
+        "If private, it will only be visible to users which have been granted a "
+        "role on the endpoint, have been granted a role on one of its collections, or "
+        "belong to a domain which has access to any of its storage gateways."
+    ),
+)
+@click.option(
+    "--subscription-id",
+    type=SubscriptionIdType(),
+    help=(
+        "'<uuid>' will set an exact subscription. 'null' will remove the current "
+        "subscription. 'DEFAULT' will instruct GCS to infer and set the subscription "
+        "from your user (requires that you are a subscription manager of exactly one "
+        "subscription)"
+    ),
+)
+@network_use_constraints
+@LoginManager.requires_login("transfer")
+def update_command(
+    login_manager: LoginManager,
+    *,
+    endpoint_id: uuid.UUID,
+    allow_udt: bool | None,
+    contact_email: str | None,
+    contact_info: str | None,
+    department: str | None,
+    description: str | None,
+    display_name: str | None,
+    gridftp_control_channel_port: str | None,
```

Comment:
I'm reading the options correctly for this, yeah?

```suggestion
    gridftp_control_channel_port: int | ExplicitNullType | None,
```

Reply:
Ah correct, good catch. New Rev.